clouddrift/binning.py

```diff
@@ -131,40 +425,70 @@ def binned_statistics(
-                where=bin_counts > 0,
-            )
-        else:
-            # if no data is provided histogram by counting the coords
-            flat_idx = np.ravel_multi_index(indices, edges_sz)
-            bin_counts = np.bincount(flat_idx, minlength=np.prod(edges_sz))
-            if zeros_to_nan and np.any(bin_counts == 0):
-                bin_counts = np.where(bin_counts == 0, np.nan, bin_counts)
-
-        # add variable(s) to dataset
-        # add bin count
-        count_var_name = f"{name}_count" if var.size else f"{name}"
-        ds[count_var_name] = xr.DataArray(
-            bin_counts.reshape(edges_sz),
-            dims=dim_names,
-            coords=dict(zip(dim_names, bin_centers)),
-        )
+        # loop through statistics for the variable
+        bin_count, bin_mean, bin_sum = None, None, None
+        for statistic in statistics:
+            if statistic == "count":
+                binned_stats = _binned_count(flat_idx, n_bins)
+                bin_count = binned_stats.copy()
+            elif statistic == "sum":
+                binned_stats = _binned_sum(flat_idx, n_bins, var_finite)
+                bin_sum = binned_stats.copy()
+            elif statistic == "mean":
+                binned_stats = _binned_mean(
+                    flat_idx,
+                    n_bins,
+                    var_finite,
+                    bin_counts=bin_count,
+                    bin_sum=bin_sum,
+                )
+                bin_mean = binned_stats.copy()
+            elif statistic == "std":
+                binned_stats = _binned_std(
+                    flat_idx,
+                    n_bins,
+                    var_finite,
+                    bin_counts=bin_count,
+                    bin_mean=bin_mean,
+                )
+            elif statistic == "min":
+                binned_stats = _binned_min(
+                    flat_idx,
+                    n_bins,
+                    var_finite,
+                    fill_value=np.nan,
+                )
+            elif statistic == "max":
+                binned_stats = _binned_max(
+                    flat_idx,
+                    n_bins,
+                    var_finite,
+                    fill_value=np.nan,
+                )
+            else:
+                binned_stats = _binned_apply_func(
+                    flat_idx,
+                    n_bins,
+                    var_finite,
+                    func=statistic,
+                    fill_value=np.nan,
+                )
 
-        # and statistic when data is provided
-        if var.size:
-            ds[name] = xr.DataArray(
-                mean.reshape(edges_sz),
+            if zeros_to_nan and np.any(binned_stats == 0):
+                binned_stats = np.where(binned_stats == 0, np.nan, binned_stats)
+
+            # and variable to the Dataset
+            variable_suffix = (
+                _get_function_name(statistic) if callable(statistic) else statistic
```

Comment:
passing the suffix "anon_lambda_func" is a bit much. Why "anon"? Maybe the suffix could be "stat0", "stat1", etc based on the number of lambda functions passed in the `statistics` input argument?

Reply:
Anon is for Anonymous lambda. I could try to figure out a clever way to add a counter. Right now the function name was retrieved only when adding the results to the dataset.

Reply:
Fixed.